src/averageAges.js

```diff
@@ -38,6 +44,11 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const filteredWomen = people.filter(person => withChildren
+    ? person.sex === 'f' && people.find(any => any.mother === person.name)
```

Comment:
Here 
 _**some  method**_
 is also useful.  Because to determine if that person is a mother, if we find anybody that mentions her name, it is enough. We don't need to find who her child is. It works perfectly here ofc. Just wanted to mention a different way also. 
Same for determining if that person is a child of somebody.

Reply:
totally agree with @nurdanokcu
also variable naming `any` is, you know 🙂 
```suggestion
    ? person.sex === 'f'
      && people.some(child => child.mother === person.name)
```